### PR TITLE
New blower FSM.

### PR DESCRIPTION
**Pull request by jlebar**
<git-pr-chain>

#### Commits in this PR
1. New blower FSM.
    
    The moment we've all been waiting for...
    
    - Create blower_fsm, which separates out controlling the pressure
      setpoint from the PID loop that drives the fan to achieve this
      setpoint.
    
    - Rename control to blower_pid now that its role is smaller.
    
    - Add setpoint field to ControllerStatus proto so the GUI can observe
      the pressure we're trying to drive.
    
    Fixes issue #17.

#### [PR chain](https://github.com/jlebar/git-pr-chain)
1. 👉 #204 New blower FSM. 👈 **YOU ARE HERE**
1. #203 Rejigger sensors constants.
1. #205 Clarify comment on sensor scaling factor.

</git-pr-chain>

### CODE REVIEW

File: controller/lib/core/blower_fsm.cpp [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+// "Breath finite state machine" for pressure control mode.
+//
+// Currently our pressure-control mode drives a simple square wave: We go from
+// PIP pressure on inhale to PEEP pressure on exhale.
+class PressureControlFsm {
+public:
+  explicit PressureControlFsm(const VentParams &params)
+      : inspire_pressure_(cmH2O(params.pip_cm_h2o)),
+        expire_pressure_(cmH2O(params.peep_cm_h2o)),
+        inspire_end_(Hal.now() + time_per_inspire(params)),
```
Comment (jkff) [inline controller/lib/core/blower_fsm.cpp:67]: Can this compute Hal.now() only once, for peace of mind?

File: controller/lib/core/blower_fsm.cpp [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+public:
+  OffFsm() = default;
+  explicit OffFsm(const VentParams &) {}
+  Pressure get_setpoint() { return kPa(0); }
+  bool finished() { return true; }
+};
+
+// "Breath finite state machine" for pressure control mode.
+//
+// Currently our pressure-control mode drives a simple square wave: We go from
```
Comment (jkff) [inline controller/lib/core/blower_fsm.cpp:60]: Is the simple square wave acceptable with a real patient? If no, please mention what it is supposed to do instead.

File: controller/lib/core/blower_fsm.cpp (hunk 1 above)
Comment (jlebar) [reply controller/lib/core/blower_fsm.cpp:67]: Done.  Means I can't use an initialization list (sad), but okay, we deal.

File: controller/lib/core/blower_fsm.cpp (hunk 2 above)
Comment (jlebar) [reply controller/lib/core/blower_fsm.cpp:60]: I think we believe it will be; added a comment.

**Pull request merged**